rotkehlchen/api/rest.py

```diff
@@ -1837,3 +1837,38 @@ def get_asset_icon(
             response.set_etag(hashlib.md5(image_data).hexdigest())
 
         return response
+
+    def _sync_data(self, name: str, action: Literal['upload', 'download']) -> Dict[str, Any]:
+        try:
+            success, msg = self.rotkehlchen.premium_sync_data(name, action)
+
+            if success is False:
+                return wrap_in_fail_result(msg, status_code=HTTPStatus.CONFLICT)
+            else:
+                return _wrap_in_ok_result({})
+        except RemoteError as e:
+            return wrap_in_fail_result(str(e), status_code=HTTPStatus.BAD_GATEWAY)
+        except PremiumApiError as e:
+            return wrap_in_fail_result(str(e), status_code=HTTPStatus.UNAUTHORIZED)
+
+    @require_premium_user(active_check=True)
```

Comment:
The active check is not required. If you check the modifier's implementation you will see this just performs a call to Rotki's server to see if the given api keys work. This will already happen inside the logic of the sync code you are implementing and if they are not okay, a remote error will be thrown. So better handle it like that and avoid an extra roundtrip to the server.